### ISSUE TITLE
Template in yaml format

### ISSUE THREAD

**Issue by kfaldu** [ukayani/ecs-service#9]
As of now it supports only `json` format CloudFormation template.  
If we can support `yaml` as well it will be awesome.

Code base
https://github.com/ukayani/ecs-service/blob/master/lib/index.js#L133

Adding utility to check for file type and convert to object
https://www.npmjs.com/package/yamljs

**Comment by ukayani** [comment]
@kfaldu , I agree it would be nice to have YAML support. However, it is not as simple as modifying one line and pulling in a YAML parsing library. I've tried to play around with YAML parsing in node and unfortunately, none of the libraries are able to correctly parse YAML which has custom tags such as the ones in a CloudFormation template (ie.  `!Ref,  !Sub, etc). Even if they could parse it correctly, they tend to erase all information about tags once they unmarshal to an object. Because of that behaviour, I wouldn't be able to marshal back to YAML once I modify the environment variables. 

Does that make sense?

I've been working with Go recently and although similar issues exist with YAML parsing libraries in Go, i've managed to modify one to support this use case. When i get some time, i'll rewrite this utility in Go to support YAML and more advanced YAML syntax.

**Comment by kfaldu** [comment]
@ukayani yup makes sense for `!Ref, !Sub` - because amazon has these crazy functions 👍 

wohooo and excited for Go :)
